core/src/toga/app.py

```diff
@@ -485,6 +485,46 @@ def current_window(self):
         """Return the currently active content window."""
         return self._impl.current_window().interface
 
+    @property
+    def maximized(self):
+        """Returns True if any of the app's windows are maximized."""
+        return any(window.maximized for window in self.windows)
+
+    @maximized.setter
+    def maximized(self, condition: bool):
+        if condition:
+            self._impl.set_maximize_screen(self.windows)
+        else:
+            self._impl.set_normal_screen(self.windows)
+
+    @property
+    def minimized(self):
+        """Returns True only if all of the app's windows are minimized."""
+        return all(window.minimized for window in self.windows)
+
+    @minimized.setter
+    def minimized(self, condition: bool):
+        if condition:
+            self._impl.set_minimize_screen(self.windows)
+        else:
+            self._impl.set_normal_screen(self.windows)
+
+    @property
+    def full_screen(self):
+        """Return True if any of the app's window are in full screen."""
+        return any(window.full_screen for window in self.windows)
+
+    @full_screen.setter
+    def full_screen(self, condition):
+        if condition:
+            self._impl.set_full_screen(self.windows)
+        else:
+            self._impl.set_normal_screen(self.windows)
+
+    def set_normal_screen(self):
+        """Sets the app's windows in normal state"""
+        self._impl.set_normal_screen(self.windows)
+
```

Comment:
Yes an app-wide window state operations is what I aimed for the API. 

But, we now have window state properties, so we can use them instead of functions like `set_maximize_screen` or `set_minimize_screen`. Hence, I have removed them to keep a clean API and provide a single way to do a thing.

If there will be `maximized`,`minimized` properties, then it's natural to have a `full_screen` property also to have an API continuity between the various windows states properties. 

`set_full_screen()` is still there as it was defined in the core API and was also specified in the docs. So, for backwards compatibility, it is still present.

Reply:
Sure - but that doesn't address "set_normal_screen()".